apps/web/src/components/views/SettingsView/components/SettingsLanding.tsx

```diff
@@ -1,50 +1,181 @@
+"use client"
 import AvatarImgChange from "@/app/settings/components/ChangeAvatar"
-import { CountryCodeSelect } from "@/app/settings/components/CountryCodeSelect"
-import { TextInput, Textarea } from "@dotkomonline/ui"
+import { trpc } from "@/utils/trpc/client"
+import type { UserWrite } from "@dotkomonline/types"
+import { Button, PhoneInput, TextInput, Textarea, isValidPhoneNumber } from "@dotkomonline/ui"
+import { zodResolver } from "@hookform/resolvers/zod"
 import type { NextPage } from "next"
 import type { User } from "next-auth"
+import { useState } from "react"
+import { useForm } from "react-hook-form"
+import { z } from "zod"
 
-interface FormInputProps {
-  title: string
-  children?: JSX.Element
-}
+const nameSchema = z
+  .string()
+  .min(2, "Navn må være minst 2 bokstaver")
+  .max(50, "Navn kan maks være 50 bokstaver")
+  .regex(/^[a-zA-ZæøåÆØÅ\s'-]+$/, "Navn kan ikke inneholde tall eller spesialtegn bortsett fra - og '")
```

Comment:
I don't think we should have a limitation on names here, first of all because regex can cause issues with names from e.g. FEIDE not being valid.

We actually don't need name-validation since we now always set the name from FEIDE, so we have access to users' legal names. Account without membership/usernames should probably only show their emails and we are likely fine with them not being able to change their names

Reply:
Agree, I don't think we should allow self-name changing since we're also storing the names in two places. More place for error

Reply:
> since we're also storing the names in two places.

What do you mean by this? The single source of truth for us should be Auth0

Reply:
I agree that this can cause issues, but what about users not registering with FEIDE? Can we force name inputs on account registration?

Reply:
Even social members will have their name set through FEIDE (and if not that is HS badly modifying data / some desperate actions related to ITEX). This is then only relevant for what are essentially guest-users, which I think has only been used for utmatrikulering/immball (?)